crypto/fipsmodule/sha/sha3.c

```diff
@@ -35,7 +69,17 @@ int SHA3_Init(KECCAK1600_CTX *ctx, uint8_t pad, size_t bit_len) {
         exit(1);
   }
 
-  size_t block_size = SHA3_BLOCKSIZE(bit_len);
+  size_t block_size;
+
+  // The block size in SHA3 depends on the digest |bit_len|
+  // The block size in SHAKE depends only on the security level
+  // SHAKE produces arbitrary length digest
+  if (pad == SHA3_PAD_CHAR) {
```

Comment:
I can add a different parameter (e.g., a bool val) to check if the function is fixed or extendable output length.  IMO, it will make the code more "clean" but it requires a change in the SHA3 (internal) API.

Reply:
differentiating based on the PAD_CHAR seems fine to me, but it'd be nice to have a slightly better explanation in the comment above.

Reply:
Perfect. Just modified the comment to make the use of the `SHA3_PAD_CHAR` more clear.